### PR TITLE
Added type filters

### PR DESCRIPTION
**Pull request by daemoon**
This is about the gist of it. Type will be generated based on codename field taken by reflection. Let me know if this is correct approach, and if anything else needs to be done.

### CODE REVIEW

File: KenticoCloud.Delivery/DeliveryClient.cs [hunk 1]
```diff
@@ -521,5 +525,24 @@ public DeliveryClient(string projectId, string previewApiKey) : this(new Deliver
 
             throw new DeliveryException(response.StatusCode, await response.Content.ReadAsStringAsync());
         }
+
+        private bool TryGetContentTypeCodename(Type contentType, out string codename)
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery/DeliveryClient.cs]: I'd definitely put the whole thing in a separate class and unit-test it.
Also, consider making it an extension method. `parameters.ApplyTypeParameters<T>()`. But that's just a matter of taste.

File: KenticoCloud.Delivery/DeliveryClient.cs [hunk 2]
```diff
@@ -247,7 +248,9 @@ public DeliveryClient(string projectId, string previewApiKey) : this(new Deliver
                 throw new ArgumentException("Entered item codename is not valid.", nameof(codename));
             }
 
-            var endpointUrl = UrlBuilder.GetItemUrl(codename, parameters);
+            var enhancedParams = ExtractParameters<T>(parameters);
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery/DeliveryClient.cs]: it's redundant here - codename is more specific than type

**Comment by daemoon** [comment]
Fixed those issues, please review

File: KenticoCloud.Delivery/DeliveryClient.cs (hunk 2 above)
Comment (daemoon) [reply KenticoCloud.Delivery/DeliveryClient.cs]: FIxed

File: KenticoCloud.Delivery/DeliveryClient.cs (hunk 1 above)
Comment (daemoon) [reply KenticoCloud.Delivery/DeliveryClient.cs]: Extracted into separate class which was unit tested. We can do it as extension method later.

**Pull request merged**